Aliases of array insert()/prepend() generate wrong c codes

### Describe the bug

Aliases of array insert()/prepend() generate wrong c codes.

### Reproduction Steps

```v
type Strings = []string

fn main() {
	mut strs := Strings(['hi', '1', '5', '3'])
	strs.insert(0, '22')
	println(strs)
	strs.prepend('33')
	println(strs)
}
```

### Expected Behavior

```v
['22', 'hi', '1', '5', '3']
['33', '22', 'hi', '1', '5', '3']
```

### Current Behavior

```v
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/t1.01J8RN2G2YKKRBED0PRA7GWETM.tmp.c:13334: error: cannot convert 'struct string' to 'void *'
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.7 b1c9941

### Environment details (OS name and version, etc.)

V full version: V 0.4.7 b1c9941
OS: linux, Deepin 20.9 (VM)
Processor: 6 cpus, 64bit, little endian, AMD Ryzen 7 6800H with Radeon Graphics

getwd: /home/yuyi/test/t1
vexe: /home/yuyi/v/v
vexe mtime: 2024-09-26 15:44:32

vroot: OK, value: /home/yuyi/v
VMODULES: OK, value: /home/yuyi/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.20.1
Git vroot status: weekly.2024.39-22-gb1c99410 (5 commit(s) behind V master)
.git/config present: true

CC version: cc (Uos 8.3.0.3-3+rebuild) 8.3.0
thirdparty/tcc status: thirdparty-linux-amd64 0134e9b9
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/builtin/array_test.v
+++ b/vlib/builtin/array_test.v
@@ -191,6 +191,24 @@ fn test_prepend_many() {
 	assert a == [5, 6, 1, 2, 3, 4]
 }
 
+type Strings = []string
+
+fn test_aliases_of_array_insert() {
+	mut strs := Strings(['hi'])
+	strs.insert(0, '22')
+	assert strs == ['22', 'hi']
+	strs.insert(1, ['11'])
+	assert strs == ['22', '11', 'hi']
+}
+
+fn test_aliases_of_array_prepend() {
+	mut strs := Strings(['hi'])
+	strs.prepend('22')
+	assert strs == ['22', 'hi']
+	strs.prepend(['44', '33'])
+	assert strs == ['44', '33', '22', 'hi']
+}
+
 fn test_strings() {
 	a := ['a', 'b', 'c']
 	assert a.str() == "['a', 'b', 'c']"

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -879,7 +879,7 @@ fn (mut g Gen) gen_array_insert(node ast.CallExpr) {
 	elem_type_str := g.typ(left_info.elem_type)
 	arg2_sym := g.table.final_sym(node.args[1].typ)
 	is_arg2_array := arg2_sym.kind == .array
-		&& node.args[1].typ.clear_flag(.variadic) == g.table.unaliased_type(node.left_type)
+		&& node.args[1].typ.clear_flag(.variadic) in [node.left_type, g.table.unaliased_type(node.left_type)]
 	noscan := g.check_noscan(left_info.elem_type)
 	addr := if node.left_type.is_ptr() { '' } else { '&' }
 	if is_arg2_array {
@@ -916,7 +916,7 @@ fn (mut g Gen) gen_array_prepend(node ast.CallExpr) {
 	elem_type_str := g.typ(left_info.elem_type)
 	arg_sym := g.table.final_sym(node.args[0].typ)
 	is_arg_array := arg_sym.kind == .array
-		&& node.args[0].typ == g.table.unaliased_type(node.left_type)
+		&& node.args[0].typ in [node.left_type, g.table.unaliased_type(node.left_type)]
 	noscan := g.check_noscan(left_info.elem_type)
 	addr := if node.left_type.is_ptr() { '' } else { '&' }
 	if is_arg_array {

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -874,12 +874,12 @@ fn (mut g Gen) gen_array_filter(node ast.CallExpr) {
 
 // `nums.insert(0, 2)` `nums.insert(0, [2,3,4])`
 fn (mut g Gen) gen_array_insert(node ast.CallExpr) {
-	left_sym := g.table.sym(node.left_type)
+	left_sym := g.table.final_sym(node.left_type)
 	left_info := left_sym.info as ast.Array
 	elem_type_str := g.typ(left_info.elem_type)
-	arg2_sym := g.table.sym(node.args[1].typ)
+	arg2_sym := g.table.final_sym(node.args[1].typ)
 	is_arg2_array := arg2_sym.kind == .array
-		&& node.args[1].typ.clear_flag(.variadic) == node.left_type
+		&& node.args[1].typ.clear_flag(.variadic) == g.table.unaliased_type(node.left_type)
 	noscan := g.check_noscan(left_info.elem_type)
 	addr := if node.left_type.is_ptr() { '' } else { '&' }
 	if is_arg2_array {
@@ -911,11 +911,12 @@ fn (mut g Gen) gen_array_insert(node ast.CallExpr) {
 
 // `nums.prepend(2)` `nums.prepend([2,3,4])`
 fn (mut g Gen) gen_array_prepend(node ast.CallExpr) {
-	left_sym := g.table.sym(node.left_type)
+	left_sym := g.table.final_sym(node.left_type)
 	left_info := left_sym.info as ast.Array
 	elem_type_str := g.typ(left_info.elem_type)
-	arg_sym := g.table.sym(node.args[0].typ)
-	is_arg_array := arg_sym.kind == .array && node.args[0].typ == node.left_type
+	arg_sym := g.table.final_sym(node.args[0].typ)
+	is_arg_array := arg_sym.kind == .array
+		&& node.args[0].typ == g.table.unaliased_type(node.left_type)
 	noscan := g.check_noscan(left_info.elem_type)
 	addr := if node.left_type.is_ptr() { '' } else { '&' }
 	if is_arg_array {

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -1582,7 +1582,7 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 	left_sym := g.table.sym(left_type)
 	final_left_sym := g.table.final_sym(left_type)
 	if left_sym.kind == .array || (final_left_sym.kind == .array
-		&& node.name in ['filter', 'map', 'sort', 'sorted', 'contains', 'any', 'all']) {
+		&& node.name in ['filter', 'map', 'sort', 'sorted', 'contains', 'any', 'all', 'insert', 'prepend']) {
 		if g.gen_array_method_call(node, left_type) {
 			return
 		}

